test/idlemonitor/DefaultIdleMonitor.test.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+
+import DefaultIdleMonitor from '../../src/idlemonitor/DefaultIdleMonitor';
+import PingPong from '../../src/pingpong/PingPong';
+import PingPongObserver from '../../src/pingpongobserver/PingPongObserver';
+import TimeoutScheduler from '../../src/scheduler/TimeoutScheduler';
+import DOMMockBuilder from '../dommock/DOMMockBuilder';
+
+describe('DefaultIdleMonitor', () => {
+  let expect: Chai.ExpectStatic;
+  let maxIdleTimeMs: number;
+  let domMockBuilder: DOMMockBuilder | null = null;
+
+  class TestPingPong implements PingPong {
+    addObserver(_observer: PingPongObserver): void {}
+
+    removeObserver(_observer: PingPongObserver): void {}
+
+    forEachObserver(_observerFunc: (_observer: PingPongObserver) => void): void {}
+
+    start(): void {}
+
+    stop(): void {}
+  }
+
+  beforeEach(() => {
+    expect = chai.expect;
```

Comment:
Could you move this and maxIdleTimeMs up? We dont need to reset this for every test.